vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java

```diff
@@ -127,21 +139,45 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
    * @param options options to configure the update
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient updateWithOptions(String collection, JsonObject query, JsonObject update, UpdateOptions options, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Update matching documents in the specified collection, specifying options and return the handler with MongoClientUpdateResult result
+   *
+   * @param collection  the collection
+   * @param query  query used to match the documents
+   * @param update used to describe how the documents will be updated
+   * @param options options to configure the update
+   * @param resultHandler will be called when complete
+   */
+  @Fluent
+  MongoClient updateWithOptionsWithMongoClientUpdateResult(String collection, JsonObject query, JsonObject update, UpdateOptions options,
+                                                           Handler<AsyncResult<MongoClientUpdateResult>> resultHandler);
```

Comment:
suggested naming change `updateCollectionWithOptions` - although as it has a different method signature to `update` you could actually call this `updateCollection` as well